hs_composite_resource/models.py

```diff
@@ -52,24 +134,299 @@ def get_metadata_xml(self, pretty_print=True, include_format_elements=True):
+            else:
+                aggregation.create_aggregation_xml_documents()
+                # if we found an aggregation by the folder name that means this folder doesn't
+                # have any sub folders as multi-file aggregation folder can't have sub folders
+        except ObjectDoesNotExist:
+            # create xml map and metadata xml documents for all aggregations that exist
+            # in *folder* and its sub-folders
+            if not folder.startswith(self.file_path):
+                folder = os.path.join(self.file_path, folder)
+
+            res_file_objects = ResourceFile.list_folder(self, folder)
+            aggregations = []
+            for res_file in res_file_objects:
+                if res_file.has_logical_file and res_file.logical_file not in aggregations:
+                    aggregations.append(res_file.logical_file)
+
+            if check_metadata_dirty:
+                aggregations = [aggr for aggr in aggregations if aggr.metadata.is_dirty]
+            for aggregation in aggregations:
+                aggregation.create_aggregation_xml_documents()
+
+    def get_aggregation_by_name(self, name):
+            """Get an aggregation that matches the aggregation name specified by *name*
+            :param  name: name of the aggregation to find
+            :return an aggregation object if found
+            :raises ObjectDoesNotExist if no matching aggregation is found
+            """
+            for aggregation in self.logical_files:
+                if aggregation.aggregation_name == name:
+                    return aggregation
+
+            raise ObjectDoesNotExist("No matching aggregation was found for name:{}".format(name))
+
+    def recreate_aggregation_xml_docs(self, orig_aggr_name, new_aggr_name):
+        """
+        When a folder or file representing an aggregation is renamed or moved,
+        the associated map and metadata xml documents are deleted
+        and then regenerated
+        :param  orig_aggr_name: original aggregation name - used for deleting existing
+        xml documents
+        :param  new_aggr_name: new aggregation name - used for finding a matching
+        aggregation so that new xml documents can be recreated
+        """
+
+        """
+        check if the following logic meets the use cases of file/folder rename/move
+        case-1 (renaming a single aggregation at the root):
+                orgi_aggr: text.txt
+                new_aggr: text_rename.txt
+                Note: needs deleting old xml files
+
+        case-2 (renaming a single aggregation inside a folder):
+                orgi_aggr: folder_1/text.txt
+                new_aggr: folder_1/text_rename.txt
+                Note: needs deleting old xml files
+
+        case-3 (renaming a folder that contains a single aggregation):
+                orgi_aggr: folder_1/text.txt
+                new_aggr: folder_2/text.txt
+                Note: no need for deleting old xml files
```

Comment:
Is this true?  Doesn't the aggregation resmap have the folder path in it?

Reply:
oh, I think I understand now.  You don't need to delete the old xml file because it will be overwritten?

Reply:
Correct